typings/browser/ambient/amqplib/index.d.ts

```diff
@@ -131,8 +131,8 @@ declare module "amqplib" {
         bindExchange(destination: string, source: string, pattern: string, args?: any): when.Promise<Replies.Empty>;
         unbindExchange(destination: string, source: string, pattern: string, args?: any): when.Promise<Replies.Empty>;
 
-        publish(exchange: string, routingKey: string, content: Buffer, options?: Options.Publish): boolean;
-        sendToQueue(queue: string, content: Buffer, options?: Options.Publish): boolean;
+        publish(exchange: string, routingKey: string, content: Buffer, options?: Options.Publish): when.Promise<any>;
```

Comment:
what's a `when.Promise`? Some 3rd party promise lib?

Reply:
Yep! 
https://github.com/cujojs/when

I tried to just use "Promise" but it wasn't working 🤔

Reply:
Should the code you added use `when` also instead of `new Promise`?

Reply:
It seems like they use `when` for typings and the signatures probably match.

Reply:
No, they use Bluebird everywhere in the code. No sign of `when`

Reply:
In that case, what does this mean? `import when = require("when");`

Reply:
It's importing the typings from a local folder if I understand correctly:
https://github.com/smooch/amqp-ts/tree/master/typings/main/ambient/when